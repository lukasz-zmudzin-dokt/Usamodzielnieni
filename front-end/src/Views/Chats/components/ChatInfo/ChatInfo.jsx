import React from 'react';
import { Row, Col, Button } from 'react-bootstrap';
import { IndexLinkContainer } from 'react-router-bootstrap';
import UserIcon from "components/UserIcon";

const ChatInfo = ({ chat, ...rest }) => {
    return (
        <Row {...rest}>
            <Col xs="auto">
                <UserIcon />
            </Col>
            <Col >
                <h5>{chat.name}</h5>
                <IndexLinkContainer to={`/chats/${chat.id}`}>
                    <Button>Pokaż szczegóły</Button>
                </IndexLinkContainer>
            </Col>
        </Row>
    )
}

export default ChatInfo
