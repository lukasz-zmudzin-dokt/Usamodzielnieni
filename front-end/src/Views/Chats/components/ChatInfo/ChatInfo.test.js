import React from "react";
import { render } from "@testing-library/react";
import ChatInfo from "./ChatInfo";
import { MemoryRouter } from 'react-router-dom';

describe('ChatInfo', () => {
    it('should render without crashing', () => {
        const chat = {
            id: "123",
            name: "Nazwa chatu"
        };

        const { container } = render(
            <MemoryRouter>
                <ChatInfo chat={chat} />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
});