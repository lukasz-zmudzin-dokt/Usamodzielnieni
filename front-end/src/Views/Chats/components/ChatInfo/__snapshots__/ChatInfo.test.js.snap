// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInfo should render without crashing 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-auto"
    >
      <svg
        height="80"
        width="80"
      >
        <circle
          cx="40"
          cy="40"
          fill="white"
          r="37"
          stroke="black"
          stroke-width="3"
        />
        <circle
          cx="40"
          cy="30"
          fill="black"
          r="10"
        />
        <path
          d="M 20 60 q 20 -30 40 0"
          fill="black"
        />
      </svg>
    </div>
    <div
      class="col"
    >
      <h5>
        Nazwa chatu
      </h5>
      <a
        class="btn btn-primary"
        href="/chats/123"
      >
        Pokaż szczegóły
      </a>
    </div>
  </div>
</div>
`;
